pokedex/pokedex.lua

```diff
@@ -608,7 +612,7 @@ function M.get_evolutions(pokemon, gender)
 	for _, species in pairs(d.into) do
 		if not M.enforce_genders() then
 			table.insert(evolutions, species)
-		elseif genders[species] == nil or (genders[species] and genders[species] == gender) then
+		elseif gender_can_evolve_to(gender, species) then
```

Comment:
Problem was this logic didn't have `(gender or M.GENDERLESS)`. Moved to a function for automatic consistency.